.github/workflows/deploy.yml

```diff
@@ -21,10 +21,29 @@ jobs:
       - name: Check if merged pull request was an automatic version bump PR
         id: isAutomatedPullRequest
         run: echo "::set-output name=IS_AUTOMERGE_PR::${{ steps.getMergedPullRequest.outputs.author == 'OSBotify' }}"
+  
+  setupGit:
+    runs-on: ubuntu-latest
+    needs: validate
+    steps:
+      - name: Decrypt Botify GPG key
+        run: cd .github/workflows && gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output OSBotify-private-key.asc OSBotify-private-key.asc.gpg
+        env:
+          LARGE_SECRET_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
+
+      - name: Import Botify GPG Key
+        run: cd .github/workflows && gpg --import OSBotify-private-key.asc
+
+      - name: Set up git for Botify
+        run: |
+          git config user.signingkey DBF63700F60F5530
+          git config commit.gpgsign true
+          git config user.name OSBotify
+          git config user.email infra+osbotify@expensify.com
 
   deployStaging:
     runs-on: ubuntu-latest
-    needs: validate
+    needs: [validate, setupGit]
```

Comment:
Have we tested this yet? I think since `setupGit` runs on a different runner (and hence a totally separate machine), the local git config changes that are made in the machine that runs `setupGit` won't propagate to the machines that run `deployStaging` and `deployProduction`.

Which is a long-winded way of saying I _think_ you need to do one of two things:

1. Just duplicate the steps from `setupGit` in `deployStaging` and `deployProduction`
1. OR (fancy, fancy), create a compound run steps action with the steps from `setupGit` to setup git for OSBotify and reuse that action in whatever jobs you need.

Reply:
I looked into #2 and it looked a bit involved, I will do 1!